rosidl_runtime_c/include/rosidl_runtime_c/message_bounds_struct.h

```diff
@@ -28,21 +28,50 @@ typedef struct rosidl_message_bounds_t rosidl_message_bounds_t;
 typedef const rosidl_message_bounds_t * (* rosidl_message_bounds_handle_function)(
   const rosidl_message_bounds_t *, const char *);
 
+/// Contains rosidl message bounds data
 struct rosidl_message_bounds_t
 {
+  /// String identifier for the type_support.
   const char * typesupport_identifier;
+  /// pointer to type support handle function
   const void * data;
+  /// pointer to type support handle function.
   rosidl_message_bounds_handle_function func;
 };
 
+/// Get the message bounds handle specific to this identifier.
+/**
+ * If the identifier is the same as this handle's identifier, then the handle is
```

Comment:
This simply returns handle->func after some casts are done (oh C). No identifier comparisons happen. And NULL return is not possible. Instead document the asserts:
https://github.com/ros2/rosidl/blob/master/rosidl_runtime_c/src/sequence_bound.c